zulipterminal/ui_tools/boxes.py

```diff
@@ -1136,6 +1136,13 @@ def soup2markup(cls, soup: Any, metadata: Dict[str, Any],
                 metadata['time_mentions'].append((time_string, source_text))
             else:
                 markup.extend(cls.soup2markup(element, metadata)[0])
+            # Remove extra newline from the end of blockquotes
+            if(markup[-1] == '\n'
+               and isinstance(markup[-2], tuple)
+               and markup[-2][0] == 'msg_quote'
+               and isinstance(markup[-2][1], list)
+               and markup[-2][1][-1] == '\n'):
+                markup = markup[:-1]
```

Comment:
This looks rather messy. Can this be included in the quote handler? If so, would it be clearer there?

Reply:
This piece of code checks for extra newlines after quotes and removes them. But, I agree, this looks messy and complex, mainly due to the following reason:
- Our `soup2mark` function is recursive, hence it becomes difficult to check for **unnecessary** blank lines after quotes.
Currently, I am relying on `markup` to check for any appended `\n` at the end of each recursion-step, then comparing it with the previous element if it was `blockquote` and had a `\n` at the end already. We then remove this extra newline from the end of markup.

Let me see if I can simplify this!